indexing-service/src/main/java/io/druid/indexing/common/index/YeOldePlumberSchool.java

```diff
@@ -113,6 +114,10 @@ public Plumber findPlumber(
     // Set of spilled segments. Will be merged at the end.
     final Set<File> spilled = Sets.newHashSet();
 
+    // IndexMerger implementation.
+    final IndexMerger theIndexMerger = config.getCustomIndexMerger() != null
```

Comment:
Regarding null handling - doesn't Jackson handle it up front, when you specify `defaultImpl=...` on the `IndexMerger` interface (or SegmentStrategy for that matter, if you apply suggestion above)?